util/topgen/intermodule.py

```diff
@@ -6,8 +6,31 @@
 import logging as log
 from typing import Dict, Tuple
 from collections import OrderedDict
+from enum import Enum
 
 from reggen.validate import check_int
+from topgen import lib
+
+IM_TYPES = ['uni', 'req_rsp']
+IM_ACTS = ['req', 'rsp', 'rcv']
+IM_CONN_TYPE = ['1-to-1', '1-to-N', 'broadcast']
+
+
+class ImType(Enum):
+    Uni = 1
+    Bi = 2
+
+
+class ImAct(Enum):
+    Req = 1
+    Rsp = 2
+    Rcv = 3
+
+
+class ImConn(Enum):
+    OneToOne = 1
+    OneToN = 2
+    Broadcast = 3
```

Comment:
hmmm...arent `Broadcast` and `OneToN` the same?

Reply:
Broadcast is OneToN but that `One` is just width 1. So spreading same signal to multiple modules. You could think the reset signal such as `sys_rst_n` is connected to multiples.

Reply:
i see...so broadcast covers cases where its like...an array of signal being sent out, while `One` is really just width 1? and Width=1 here does not necessarily mean 1 bit signals right? Rather 1 entry in the struct?

Reply:
Opposite... Humm another wrong naming here..                                                                                       
                                                                                                                                   
- OneToOne : Requster and responder width are same                                                                                 
- OneToN : Requester Width N and responders width 1                                                                                
- Broadcast : Requester Width 1 and responders width 1

Reply:
ah you should just put in that as a comment! For broadcast do you mean Requester Width 1 and responders with N?

Reply:
I think the naming is ok as long as there is a comment for it as suggested by Tim ;). 

Quick question: by width=1 do you mean the amount of elements in the struct array? I.e., would we be able to broadcast a 3 bit signal if those 3 bits are encapsulated within a single struct? 

Just asking, because some of the LC broadcast enables will likely fall into that category...

Reply:
Yes. At this moment, I think it needs struct if multibit of logic is connected.